src/charm.py

```diff
@@ -1,88 +1,142 @@
 
     def __init__(self, *args):
         super().__init__(*args)
-        self._container_name = "kratos-ui"
-        self.service_patcher = KubernetesServicePatch(
-            self, [("api", 3000)]
-        )
+        self._container_name = "login_ui"
+        self._container = self.unit.get_container(self._container_name)
+        """New vars should come here"""
+
+        self.unit.open_port('tcp', int(self.config.get("port")))
         self.ingress = IngressPerAppRequirer(
             self,
             relation_name="ingress",
-            port=3000,
+            port=self.config.get("port"),
             strip_prefix=True,
         )
 
-        self.framework.observe(self.on.kratos_ui_pebble_ready, self._on_kratos_ui_pebble_ready)
+        self.framework.observe(self.on.login_ui_pebble_ready, self._on_login_ui_pebble_ready)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.ingress.on.ready, self._on_ingress_ready)
+        self.framework.observe(self.ingress.on.revoked, self._on_ingress_revoked)
 
-    @property
-    def _container(self):
-        return self.unit.get_container(self._container_name)
-
-    def _on_kratos_ui_pebble_ready(self, event):
+    def _on_login_ui_pebble_ready(self, event: WorkloadEvent) -> None:
         """Define and start a workload using the Pebble API.
         """
-        self._container.add_layer("kratos_ui", self._pebble_layer, combine=True)
-        self._container.replan()
-        self.unit.status = ActiveStatus()
+        self._handle_status_update_config(event)
 
-    def _on_config_changed(self, event):
+    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         """Handle changed configuration.
         """
-        # Fetch the new config value
-        log_level = self.model.config["log-level"].lower()
+        self._handle_status_update_config(event)
+
+    def _handle_status_update_config(self, event: HookEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Login_UI container. Deferring the event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Login_UI container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuration in progress")
+
+        self._container.add_layer(self._container_name, self._login_ui_layer, combine=True)
+        logger.info("Pebble plan updated with new configuration, replanning")
+        try:
+            self._container.replan()
+        except ChangeError as err:
+            logger.error(str(err))
+            self.unit.status = BlockedStatus("Failed to replan, please consult the logs")
+            return
+
+        if not self._service_is_created:
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting for Login_UI service")
+            logger.info("Login_UI service is absent. Deferring the event.")
+            return
```

Comment:
I am not sure if this is possible to happen, I would suggest to remove this unless I am missing something